src/julia_internal.h

```diff
@@ -264,10 +265,63 @@ void jl_set_base_ctx(char *__stk);
 void jl_init_threading(void);
 void jl_start_threads(void);
 void jl_shutdown_threading(void);
+
+// Whether the GC is running
+extern char *jl_safepoint_pages;
+static inline int jl_addr_is_safepoint(uintptr_t addr)
```

Comment:
I fixed this one. However, there are a lot of threading utils that relies on c++ when compiled with MSVC. Alternative would be using C11 generic macro, which I think is even less likely going to be supported by MSVC....